lib/index.js

```diff
@@ -133,21 +133,20 @@ module.exports = function(options) {
     grunt.util.spawn(spawnOptions, (error, output) => {
       const combinedOutput = output.stdout + output.stderr
 
-      if (error) {
-        error.output = combinedOutput
-        return reject(error)
-      }
-
       if (options.save) {
-        fs.writeFile(options.save, output.stdout, error => {
+        fs.writeFile(options.save, combinedOutput, error => {
           if (error) {
             return reject(error)
           }
-          resolve(combinedOutput)
         })
-      } else {
-        resolve(combinedOutput)
       }
+
+      if (error) {
+        error.output = combinedOutput
+        return reject(error)
+      }
+
+      resolve(combinedOutput)
```

Comment:
I think we'll want to wait for the file to finish writing before resolving the Promise, otherwise file writing errors could get swallowed due to the Promise already having been settled.